### PR TITLE
Increase strictness of type checker

### PR DESCRIPTION
**Pull request by trickeydan**
- Brings type checker in line with other kit team projects
- Updates code so that all function definitions are typed
- Some type stubs are commented out, we are not using them and it'd be a pain to work them all out for no benefit.

Also a couple of other small things, please see commit messages for more information.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,13 +3,13 @@
 PYMODULE:=april_vision
 TESTS:=tests
 
-all: lint type build
+all: type lint isort-check build
```
Comment (WillB97) [inline Makefile:6]: I prefer to run linting and isort first as these issues are generally readily solved
```suggestion
all: lint isort-check type build
```

File: april_vision/cli/live.py [hunk 2]
```diff
@@ -51,6 +52,7 @@ def main(args: argparse.Namespace):
     while True:
         frame = cam._capture()
         markers = cam._detect(frame)
+        print(markers)
```
Comment (WillB97) [inline april_vision/cli/live.py:55]:
```suggestion
```

File: april_vision/cli/live.py [hunk 3]
```diff
@@ -82,7 +84,7 @@ def main(args: argparse.Namespace):
                         text_colour=(255, 191, 0),  # deep sky blue
                     )
             except RuntimeError:
-                pass
+                raise
```
Comment (WillB97) [inline april_vision/cli/live.py:87]: This is a change of behaviour.
```suggestion
                pass
```

File: april_vision/frame_sources.py [hunk 4]
```diff
@@ -5,19 +5,23 @@
 from typing import List, Optional, Tuple, Union
 
 import cv2
+import numpy as np
 from numpy.typing import NDArray
 
 LOGGER = logging.getLogger(__name__)
 
 
+FrameArray = NDArray[np.generic]
```
Comment (WillB97) [inline april_vision/frame_sources.py:14]: Facading the numpy array nature of the frames obscures the fact that they are in fact arrays and standard array and opencv methods can be used on them.

Comment (WillB97) [inline april_vision/frame_sources.py:14]: Facading the numpy array nature of the frames obscures the fact that they are in fact arrays and standard array and opencv methods can be used on them.

File: april_vision/marker.py [hunk 5]
```diff
@@ -221,7 +223,7 @@ class Orientation:
         [0, 0, -1],
     ]))
 
-    def __init__(self, rotation_matrix: NDArray, aruco_orientation: bool = True):
+    def __init__(self, rotation_matrix: NDArray[np.int8], aruco_orientation: bool = True):
```
Comment (WillB97) [inline april_vision/marker.py:226]: I'm fairly sure that the rotation matrix is floating point.

File: stubs/libusb_package/__init__.pyi [hunk 6]
```diff
@@ -32,4 +33,4 @@ def find_library(candidate: str) -> Optional[str]: ...
 def get_libusb1_backend() -> Optional["IBackend"]: ...
 
 
-def find(*args: Any, **kwargs: Any) -> Any: ...
+def find(find_all: bool, custom_match: Optional[Callable[[usb.core.Device], bool]] = None) -> Iterable[usb.core.Device]: ...
```
Comment (WillB97) [inline stubs/libusb_package/__init__.pyi:36]: I don't think we should be changing this type signature. This was explicitly set in the library itself.

**Review by WillB97 — COMMENTED** [review]
I also want to remain with relative imports withing the library.

**Comment by trickeydan** [comment]
> I also want to remain with relative imports withing the library.

Please explain why you want to break convention? We should be making things as standard as possible to reduce onboarding time for volunteers to this project.

File: stubs/libusb_package/__init__.pyi (hunk 6 above)
Comment (trickeydan) [reply stubs/libusb_package/__init__.pyi:36]: This is a type stub that ensures we are passing valid arguments. It doesn't change the behaviour of the library, it just ensures that we can be sure of the types we are dealing with. 

Adding stubs over unspecified arguments isn't just standard, it's recommended best practice.

File: april_vision/frame_sources.py (hunk 4 above)
Comment (trickeydan) [reply april_vision/frame_sources.py:14]: Sorry I don't understand how reducing repetition of this type obscures anything. It's still the same type.

**Comment by WillB97** [comment]
Superseded by #16

**Pull request closed without merging**